### PR TITLE
Effect Hitmark Ok

### PR DESCRIPTION
**Pull request by hensldm**

### CODE REVIEW

File: include/functions.h [hunk 1]
```diff
@@ -553,9 +553,9 @@ void EffectSsHahen_SpawnBurst(PlayState* play, Vec3f* pos, f32 burstScale, s16 u
 // void EffectSsSibuki_Spawn(UNK_TYPE1 param_1, UNK_TYPE1 param_2, UNK_TYPE1 param_3, UNK_TYPE1 param_4, UNK_TYPE2 param_5, UNK_TYPE2 param_6, UNK_TYPE2 param_7);
 void EffectSsSibuki_SpawnBurst(PlayState* play, Vec3f* pos);
 void EffectSsStone1_Spawn(PlayState* play, Vec3f* pos, s32 reg0);
-// void EffectSsHitMark_Spawn(UNK_TYPE4 uParm1, UNK_TYPE4 uParm2, UNK_TYPE2 uParm3, Vec3f* pzParm4);
-void EffectSsHitMark_SpawnFixedScale(PlayState* play, s32 type, Vec3f* pos);
-void EffectSsHitMark_SpawnCustomScale(PlayState* play, s32 type, s16 scale, Vec3f* pos);
+// void EffectSsHitmark_Spawn(UNK_TYPE4 uParm1, UNK_TYPE4 uParm2, UNK_TYPE2 uParm3, Vec3f* pzParm4);
```
Comment (EllipticEllipsis) [inline include/functions.h:556]: Looks like this not required externally, so can be removed from here.

**Review by EllipticEllipsis — APPROVED** [review]

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c [hunk 2]
```diff
@@ -1,27 +1,128 @@
...
+u32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsHitmark_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsHitmark_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
+static Color_RGB8 sColors[] = {
+    { 255, 255, 255 }, { 255, 255, 0 }, { 255, 255, 255 }, { 255, 0, 0 },   { 255, 200, 100 }, { 200, 150, 0 },
+    { 150, 100, 0 },   { 100, 50, 0 },  { 255, 255, 255 }, { 255, 0, 0 },   { 255, 255, 0 },   { 255, 0, 0 },
+    { 255, 255, 255 }, { 0, 255, 200 }, { 255, 255, 255 }, { 150, 0, 255 },
+};
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:30]:
```suggestion
static Color_RGB8 sColors[] = {
    { 255, 255, 255 }, { 255, 255, 0 }, // EFFECT_HITMARK_WHITE Init
    { 255, 255, 255 }, { 255, 0, 0 }, // EFFECT_HITMARK_WHITE Update
    { 255, 200, 100 }, { 200, 150, 0 }, // EFFECT_HITMARK_DUST Init
    { 150, 100, 0 },   { 100, 50, 0 },  // EFFECT_HITMARK_DUST Update
    { 255, 255, 255 }, { 255, 0, 0 },   // EFFECT_HITMARK_RED Init
    { 255, 255, 0 },   { 255, 0, 0 }, // EFFECT_HITMARK_RED Update
    { 255, 255, 255 }, { 0, 255, 200 },  // EFFECT_HITMARK_METAL Init
    { 255, 255, 255 }, { 150, 0, 255 }, // EFFECT_HITMARK_METAL Update
};
```
I think there's something we can do here to group these colors a bit better. They can be grouped in pairs (primColor and envColor), and by type, but also how they're used initially vs what they're set to in update. Do you know why there are different colors or how they work?

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c [hunk 3]
```diff
@@ -1,27 +1,128 @@
...
+    SkinMatrix_SetScale(&mfScale, scale, scale, 1.0f);
+    SkinMatrix_MtxFMtxFMult(&mfTrans, &play->billboardMtxF, &mfTransBillboard);
+    SkinMatrix_MtxFMtxFMult(&mfTransBillboard, &mfScale, &mfResult);
+    gSPMatrix(POLY_XLU_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &mfResult);
+
+    if (mtx != NULL) {
+        gSPMatrix(POLY_XLU_DISP++, mtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        tex = sTextures[(this->rType * 8) + (this->rTexIndex)];
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:100]:
```suggestion
        tex = sTextures[this->rType][this->rTexIndex];
```
Could this match as a 2D array?
`static TexturePtr sTextures[4][8] = {`

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c [hunk 4]
```diff
@@ -1,27 +1,128 @@
...
+
+    colorIndex = initParams->type * 4;
+    this->rTexIndex = 0;
+    this->rType = initParams->type;
+    this->rPrimColorR = sColors[colorIndex].r;
+    this->rPrimColorG = sColors[colorIndex].g;
+    this->rPrimColorB = sColors[colorIndex].b;
+    this->rEnvColorR = sColors[colorIndex + 1].r;
+    this->rEnvColorG = sColors[colorIndex + 1].g;
+    this->rEnvColorB = sColors[colorIndex + 1].b;
```
Comment (engineer124) [inline src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:70]:
```suggestion
    this->rPrimColorR = sColors[colorIndex][0].r;
    this->rPrimColorG = sColors[colorIndex][0].g;
    this->rPrimColorB = sColors[colorIndex][0].b;
    this->rEnvColorR = sColors[colorIndex][1].r;
    this->rEnvColorG = sColors[colorIndex][1].g;
    this->rEnvColorB = sColors[colorIndex][1].b;
```
Could this also match as a 2D array?
`static Color_RGB8 sColors[8][2] = {

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c (hunk 3 above)
Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:100]: Unfortunately, No. Same as in OoT.

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c (hunk 2 above)
Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:30]: To be honest haven't looked to much into it. Currently I kind of want to get all the effects decompiled into main to see what is the best way to document them, then properly sync them up with OoT as well (assuming no else has plans to).

Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:30]: Currently most if not all documentation is just taken from OoT.

File: src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c [hunk 5]
```diff
@@ -1,27 +1,128 @@
...
+};
+
+static TexturePtr sTextures[] = {
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,  gEffHitmark9Tex,  gEffHitmark10Tex, gEffHitmark11Tex, gEffHitmark12Tex,
+    gEffHitmark13Tex, gEffHitmark14Tex, gEffHitmark15Tex, gEffHitmark16Tex, gEffHitmark17Tex, gEffHitmark18Tex,
+    gEffHitmark19Tex, gEffHitmark20Tex, gEffHitmark21Tex, gEffHitmark22Tex, gEffHitmark23Tex, gEffHitmark24Tex,
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,
+};
```
[the remark this answers is not in the document]
Comment (hensldm) [reply src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c:39]: Yeah, I think those would be better names. Though perhaps `gEffHitmarkWhiteMetal#Tex` for the first group.

**Review by engineer124 — APPROVED** [review]

**Pull request merged**